[docs] Add clarification about reactivity and arrays

stephane-vanraes opened the pull request:
Because this is a common recurring question, this PR adds a touch more information in regards to reactivity and arrays.  More specifically the need for an assignment after array manipulation.

This is already covered in the tutorial, but it seems good to also include it as an example directly in the docs themselves.

dummdidumm commented on site/content/docs/01-component-format.md:
```diff
@@ -95,7 +95,7 @@ To change component state and trigger a re-render, just assign to a locally decl
 Update expressions (`count += 1`) and property assignments (`obj.x = y`) have the same effect.
 
-Because Svelte's reactivity is based on assignments, using array methods like `.push()` and `.splice()` won't automatically trigger updates. Options for getting around this can be found in the [tutorial](tutorial/updating-arrays-and-objects).
+Because Svelte's reactivity is based on assignments, using array methods like `.push()` and `.splice()` won't automatically trigger updates, to trigger the update a subsequent assignment is requirred. This and more details can also be found in the [tutorial](tutorial/updating-arrays-and-objects).
```
```suggestion
Because Svelte's reactivity is based on assignments, using array methods like `.push()` and `.splice()` won't automatically trigger updates. To trigger the update a subsequent assignment is required. This and more details can also be found in the [tutorial](tutorial/updating-arrays-and-objects).
```

dummdidumm reviewed (commented):
Makes sense, thank you! I would move the example what does and what doesnt trigger the update into the existing code block

stephane-vanraes commented:
I am not sure about moving it in the same block, I think keeping them separate avoids confusion because it makes the distinction betweeen the **normal** way and the **special** case

geoffrich commented on site/content/docs/01-component-format.md:
```diff
@@ -109,6 +109,20 @@ Because Svelte's reactivity is based on assignments, using array methods like `.
+		// this method call does not trigger an update
+		arr.push(2);
+		// this assignment will trigger an update
+		// if the markup references `count`
```
I think the wrong variable is mentioned in the comment
```suggestion
		// if the markup references `arr`
```

benmccann commented on site/content/docs/01-component-format.md:
```diff
@@ -95,7 +95,7 @@ To change component state and trigger a re-render, just assign to a locally decl
 Update expressions (`count += 1`) and property assignments (`obj.x = y`) have the same effect.
 
-Because Svelte's reactivity is based on assignments, using array methods like `.push()` and `.splice()` won't automatically trigger updates. Options for getting around this can be found in the [tutorial](tutorial/updating-arrays-and-objects).
+Because Svelte's reactivity is based on assignments, using array methods like `.push()` and `.splice()` won't automatically trigger updates. To trigger the update a subsequent assignment is required. This and more details can also be found in the [tutorial](tutorial/updating-arrays-and-objects).
```
I think you need to either add a comma or, perhaps better, reverse the sentence:

"To trigger the update, a subsequent assignment is required" or "A subsequent assignment is required to trigger the update"

benmccann commented on site/content/docs/01-component-format.md:
```diff
@@ -109,6 +109,20 @@ Because Svelte's reactivity is based on assignments, using array methods like `.
+		// this method call does not trigger an update
+		arr.push(2);
+		// this assignment will trigger an update
+		// if the markup references `arr`
```
I wonder if this should go on the previous line? it's sort of a short line break. Probably fine either way though

stephane-vanraes commented:
Fixed the language following Ben's comment.
Moved the array text and sample to their own 'row' so that it's clearer which part of the text belongs to each code block.
